homeassistant/components/lgthinq/coordinator.py

```diff
@@ -0,0 +1,225 @@
+
+        # Full response into the device api.
+        self.device_api.set_status(data)
+        self.is_connected = True
+        return data
+
+    def __str__(self) -> str:
+        """Return a string expression."""
+        return (
+            f"Coordinator:{self.device_name}"
+            f"(type={self.device_api.device_type}, id={self.device_api.device_id})"
+        )
+
+
+async def async_setup_device_coordinator(
+    hass: HomeAssistant, thinq_api: ThinQApi, device: dict[str, Any]
+) -> list[DeviceDataUpdateCoordinator] | None:
+    """Create DeviceDataUpdateCoordinator and device_api per device."""
+    device_id = device.get("deviceId")
+    if not device_id:
+        _LOGGER.error("Failed to setup device: no device id")
+        return None
+
+    device_info = device.get("deviceInfo")
+    if not device_info:
+        _LOGGER.error("Failed to setup device(%s): no device info", device_id)
+        return None
+
+    # Get an appropriate class constructor for the device type.
+    device_type = device_info.get("deviceType")
+    constructor = DEVICE_TYPE_API_MAP.get(device_type)
+    if constructor is None:
+        _LOGGER.error(
+            "Failed to setup device(%s): not supported device. type=%s",
+            device_id,
+            device_type,
+        )
+        return None
+
+    # Get a device profile from the server.
+    try:
+        profile = await thinq_api.async_get_device_profile(device_id)
+    except ThinQAPIException:
+        _LOGGER.warning("Failed to setup device(%s): no profile", device_id)
+        return None
+
+    device_group_id = device_info.get("groupId")
+
+    # Create new device api instance.
+    device_api: ConnectBaseDevice = (
+        constructor(
+            thinq_api=thinq_api,
+            device_id=device_id,
+            device_type=device_type,
+            model_name=device_info.get("modelName"),
+            alias=device_info.get("alias"),
+            group_id=device_group_id,
+            reportable=device_info.get("reportable"),
+            profile=profile,
+        )
+        if device_group_id
+        else constructor(
+            thinq_api=thinq_api,
+            device_id=device_id,
+            device_type=device_type,
+            model_name=device_info.get("modelName"),
+            alias=device_info.get("alias"),
+            reportable=device_info.get("reportable"),
+            profile=profile,
+        )
+    )
+
+    # Create a list of sub-devices from the profile.
+    # Note that some devices may have more than two device profiles.
+    # In this case we should create multiple lg device instance.
+    # e.g. 'WashTower-Single-Unit' = 'WashTower{dryer}' + 'WashTower{washer}'.
+    device_sub_ids = (
+        list(profile.keys())
+        if device_type == DeviceType.WASHTOWER and "property" not in profile
+        else [NONE_KEY]
+    )
+
+    # Create new device coordinator instances.
+    coordinator_list: list[DeviceDataUpdateCoordinator] = []
```

Comment:
Can you elaborate on when this happens and what this means?

Reply:
Because a washtower has two devices(dryer, washer), it needs two coordinators.
Because we assume a device is with one coordinator.